### PR TITLE
Stop reporting volume limits for migrated in-tree drivers in Node object

### LINKED ISSUE #113982: Stop reporting volume limits for migrated volume plugins in Node object
### What happened?

Reporting volume limits for migrated volume plugins with often incorrect values confuses users because intree limits are mostly hardcoded and are not getting updated now that CSI driver is handling these volume types.

### What did you expect to happen?

Should not report any volume limits.

### How can we reproduce it (as minimally and precisely as possible)?

Kubelet reports incorrect volume limits:

```
  61   │   capacity:
  62   │     attachable-volumes-aws-ebs: "39"
  63   │     cpu: "4"
  64   │     ephemeral-storage: 125293548Ki
  65   │     hugepages-1Gi: "0"
  66   │     hugepages-2Mi: "0"
  67   │     memory: 16129400Ki
  68   │     pods: "250"
```

### Anything else we need to know?

_No response_

### Kubernetes version

1.25

### Cloud provider

Any

### OS version

_No response_

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by mpatlasov**
#### What type of PR is this?

/kind bug
/sig storage

#### What this PR does / why we need it:

Currently volume limits are reported both in Node and in CSINode:
```
$ oc get no <node-name> -o yaml
...
status:
  allocatable:
    attachable-volumes-aws-ebs: "39"

$ oc get csinodes <node-name> -o yaml
...
spec:
  drivers:
  - allocatable:
      count: 25
```
The value reported in Node object is incorrect in confusing. PR stops reporting it.

#### Which issue(s) this PR fixes:

Fixes #113982

#### Special notes for your reviewer:

/cc @gnufied

#### Does this PR introduce a user-facing change?

```release-note
Volume limits for migrated drivers (aws_ebs, gce_pd, azure_dd) are not reported in Node object anymore.
The limits to be seen in CSINode.spec.drivers.allocatable.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```
N/A
```

### CODE REVIEW

**Comment by jsafrane** [comment]
/lgtm
cc @xing-yang @msau42 for awareness. The limits would disappear from Node when we remove the volume plugins anyway.

**Comment by msau42** [comment]
I was thinking about the autoscaling case because the cluster autoscaler does not simulate CSINode objects.

Looking at https://github.com/kubernetes/kubernetes/tree/master/pkg/scheduler/framework/plugins/nodevolumelimits, if CSINode is not found, then we would fall back to Node limits. If Node limits are not found, then we treat it as no limit (infinite).

That is the same behavior you would get today if you used CSI directly. It potentially causes the [autoscaler to create an unschedulable node](https://github.com/kubernetes/kubernetes/issues/95911#issuecomment-718118538) if the volume limit is very low. 

I suppose the current behavior of using falling back to node limits can also cause incorrect autoscaler behavior, so we're trading one suboptimal behavior for another.

**Comment by mpatlasov** [comment]
> I was thinking about the autoscaling case because the cluster autoscaler does not simulate CSINode objects.

Michelle, can you give some details about autoscaler, may be some links to the code sources. Does it simulate Node objects? If yes, how it knows whether they are "Nitro" or not for AWS case? (`DefaultMaxEBSVolumes = 39` but `DefaultMaxEBSNitroVolumeLimit = 25`, and both differs from what CSI driver calculates).
@msau42

**Comment by msau42** [comment]
Here is a high level description of how autoscaler works: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-does-scale-up-work

It basically makes a copy of an existing Node object in the same "group", and then runs the same scheduling algorithms against that simulated Node. For the "scale-from-zero" case, a Node template would have to be provided with the appropriate parameters.

To summarize the change in behavior using your original example:

Before this change, autoscaler would simulate new nodes with `volume limit = 39`. After this change, autoscaler would simulate new nodes with `volume limit = infinite`.

Both are wrong, but increasing the limit to infinite could potentially cause autoscaler to underscale more frequently.

**Comment by jsafrane** [comment]
So, autoscaler is already wrong when a cluster uses only CSI driver and no in-tree volumes, right? I would prefer to adapt the autoscaler to simulate CSINode, not keeping the old (and wrong) values in Node forever.

**Comment by msau42** [comment]
Agree that the right direction is to make autoscaler aware of CSINode. But it's probably a signficant project so will take time. Over the next 2 releases, we plan to remove the intree plugins. Do we take the hit? This issue mainly impacts plugins where the per-node limits are low and easily to exceed.

**Comment by xing-yang** [comment]
AWS and Azure Disk in-tree driver are already removed.  GCE in-tree driver is targeted for removal in 1.28.

**Comment by mpatlasov** [comment]
@xing-yang Can you please advise, do you recommend to merge this PR or we have to wait for 1.28?

**Comment by jingxu97** [comment]
what is suggestion for this PR? @xing-yang @msau42

**Comment by msau42** [comment]
All 3 of these drivers are removed as of 1.28 so I don't think there is any code to remove anymore. The only consideration left is solving the autoscaling problem.
